lib/commands/serve.js

```diff
@@ -107,15 +107,25 @@ module.exports = Command.extend({
   },
 
   async _checkExpressPort(commandOptions) {
-    let foundPort = await getPort({ port: commandOptions.port, host: commandOptions.host });
-    if (commandOptions.port !== foundPort && commandOptions.port !== 0) {
-      let message = `Port ${commandOptions.port} is already in use.`;
+    let portOptions = { port: commandOptions.port, host: commandOptions.host };
+    if (commandOptions.port !== 0) {
+      portOptions.stopPort = commandOptions.port;
+    }
+    try {
+      let foundPort = await getPort(portOptions);
+      commandOptions.port = foundPort;
+      commandOptions.liveReloadPort = commandOptions.liveReloadPort || commandOptions.port;
+      return commandOptions;
+    } catch (err) {
+      let message;
+      if (commandOptions.port === 0) {
+        message = `No open port found above ${commandOptions.port}`;
+      } else if (commandOptions.port < 1024) {
+        message = `Port ${commandOptions.port} is already in use or you do not have permissions to use this port.`;
+      } else {
+        message = `Port ${commandOptions.port} is already in use.`;
+      }
       return Promise.reject(new SilentError(message));
```

Comment:
Apologies for not spotting this initially (and I know that you didn't introduce this), but we shouldn't return a rejected promise here from within an async function. We should just throw instead.

```suggestion
      throw new SilentError(message);
```

Reply:
Sure, I'll make this change and update.

Reply:
@rwjblue I have made the change but I would like to understand the rationale behind throwing an error here. My understanding is that return promise would have the same behavior as throwing an error.

Reply:
Yep! The resulting behavior is the same (the promise that invoking the outer function returns would reject). It really is just about writing idiomatic code to me.